### ISSUE TITLE
Watcher panicked after editing file

### ISSUE THREAD

**Issue by matheus-consoli** [spacedriveapp/spacedrive#2458]
### Describe the bug

The application crashes after editing a new file create from outside of Spacedrive.

### Reproduction

1. create a new Location
2. let it open on the side
3. create a new file in the location using another application (`touch file`)
4. edit the file (`vim file`)

### Expected behavior

The application should not crash

### Platform and versions

```Shell
>> pnpm --version && cargo --version && rustc --version
9.0.6
cargo 1.80.0-nightly (6087566b3 2024-04-30)
rustc 1.80.0-nightly (79734f1db 2024-05-02)
```

### Stack trace

```Shell
2024-05-06T22:37:54.692469Z ERROR sd_core: core/src/lib.rs:239: panicked at core/src/location/manager/watcher/utils.rs:942:13:
not yet implemented: file has changed in some way, re-identify it panic.file="core/src/location/manager/watcher/utils.rs:942" panic.column=13
fatal runtime error: failed to initiate panic, error 3
```

### Additional context

The issue's origin is this `todo!()`: https://github.com/spacedriveapp/spacedrive/blob/44478207e72495b3777e294660d78939711b544f/core/src/location/manager/watcher/utils.rs#L942

I'll be happy to work on it, but I will just need some help with the expected behavior :)

**Comment by fogodev** [comment]
Yeah, a concurrency issue that I mapped with this `todo!` but was hoping that it would never happen ): 

Will give it a better look when possible

**Comment by jamiepine** [comment]
Closing due to v2 rewrite. If you encounter this issue in v2, please open a new issue.
